apps/prairielearn/src/api/trpc/routers/course-files/batchDeleteQuestions.ts

```diff
@@ -1,35 +1,30 @@
 import { z } from 'zod';
 
-import { IdSchema } from '../../../../lib/db-types.js';
+import { loadSqlEquiv, queryRows } from '@prairielearn/postgres';
+
+import { IdSchema, QuestionSchema } from '../../../../lib/db-types.js';
 import { QuestionDeleteEditor } from '../../../../lib/editors.js';
 import { selectCourseById } from '../../../../models/course.js';
-import { selectQuestionById } from '../../../../models/question.js';
 import { privateProcedure, selectUsers } from '../../trpc.js';
 
-export const deleteQuestion = privateProcedure
+const sql = loadSqlEquiv(import.meta.url);
+
+export const batchDeleteQuestions = privateProcedure
   .input(
     z.object({
       // Context.
       course_id: IdSchema,
       user_id: IdSchema,
       authn_user_id: IdSchema,
       has_course_permission_edit: z.boolean(),
-
-      // Question data.
-      question_id: z.string(),
+      question_ids: z.array(IdSchema),
     }),
   )
   .output(
-    z.union([
-      z.object({
-        status: z.literal('success'),
-        job_sequence_id: z.string(),
-      }),
-      z.object({
-        status: z.literal('error'),
-        job_sequence_id: z.string(),
-      }),
-    ]),
+    z.object({
+      status: z.union([z.literal('success'), z.literal('error')]),
+      job_sequence_id: z.string(),
+    }),
```

Comment:
Why was this changed? This loses the valuable specificity of the previous types. Look into discriminated unions, e.g. https://dev.to/darkmavis1980/what-are-typescript-discriminated-unions-5hbb

Reply:
Done in 253c7cd.